src/amg/azext_amg/_params.py

```diff
@@ -43,6 +43,10 @@ def load_arguments(self, _):
         c.argument("data_source", help="name, id, uid which can identify a data source. CLI will search in the order of name, id, and uid, till finds a match")
         c.argument("definition", help="json string with data source definition, or a path to a file with such content")
 
+    with self.argument_context("grafana notification-channel") as c:
+        c.argument("notification_channel", help="id, uid which can identify a data source. CLI will search in the order of id, and uid, till finds a match")
+        c.argument("definition", help="json string with notification channel definition, or a path to a file with such content")
```

Comment:
You can use `type=validate_file_or_dict` for json string or json file. https://github.com/Azure/azure-cli-extensions/blob/81dd55b46e58078b172200825be81c1acc44f52f/src/datafactory/azext_datafactory/generated/_params.py#L395-L400

Reply:
Currently, none of the commands have type validation in place. Would you like me to expand my scope slightly and add type validation to all pre-existing commands as well?

Reply:
Yes, please. BTW, we usually don't need to specify `type` for most simple arguments.

Reply:
Applied in commits [REV2]

Type validation of the argument was added, leading to improved error messaging but also to conflict with the _try_load_file_content function currently executing type validation. This function was therefore omitted for all cli arguments except 'az grafana dashboard import' as this command takes int, str and json objects.